src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaFormulaCreator.java

```diff
@@ -564,11 +576,44 @@ public Object convertValue(Term term) {
     throw new AssertionError("Unknown value type.");
   }
 
-  public void addVariableCast(Term equal) {
-    variableCasts.add(equal);
+  public void addVariableCast(String newVariable, Term equal) {
+    variableCasts.put(newVariable, equal);
   }
 
-  public Iterable<Term> getVariableCasts() {
-    return variableCasts;
+  private Map<String, Set<String>> calculateTransitions(Map<String, Term> pVariableCasts) {
+    return Maps.transformValues(
+        pVariableCasts,
+        term ->
+            Sets.intersection(
+                pVariableCasts.keySet(), extractVariablesAndUFs(term, false).keySet()));
+  }
+
+  private Set<String> initialSet(Iterable<Term> pTerms) {
+    ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+    for (Term term : pTerms) {
+      builder.addAll(extractVariablesAndUFs(term, false).keySet());
+    }
+    return builder.build();
+  }
+
+  private Set<String> takeAStep(Map<String, Set<String>> pTransitions, Set<String> pVariables) {
+    ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+    for (String var : pVariables) {
+      builder.addAll(pTransitions.get(var));
+    }
+    return builder.build();
+  }
+
+  public Iterable<Term> getVariableCasts(Iterable<Term> pTerms) {
```

Comment:
What does this code? This needs to be commented and explained.

Reply:
Looks like a fixedpoint iteration that collects variables from terms, and from related terms.

Reply:
> Looks like a fixedpoint iteration that collects variables from terms, and from related terms, and from their related terms,..

Yes, that's the idea. Whenever a formula is pushed onto the assertion stack we need to check if it uses any variables that come from fp-to-bv casts and then include side-conditions for those casts. This needs to be done recursively until all casts are covered.

I think there are a few other solvers (like CVC5) that don't allow fp-to-bv casting?  It would be possible to include this code in AbstractBitformulaManager so that it than can be used by those solvers as well.